c7n/commands.py

```diff
@@ -232,12 +236,32 @@ def validate(options):
                 )
             ))
         used_policy_names = used_policy_names.union(conf_policy_names)
+        source_locator = None
+        if fmt[0] == "y":
+            # For yaml files there is at least the expectation that the policy
+            # name is on a line by itself. With JSON, the file could be one big
+            # line. At this stage we are only attempting to find line number for
+            # policies in yaml files.
+            source_locator = SourceLocator(config_file)
         if not errors:
             null_config = Config.empty(dryrun=True, account_id='na', region='na')
             for p in data.get('policies', ()):
                 try:
                     policy = Policy(p, null_config, Bag())
                     policy.validate()
+                    # If the policy is invalid, there isn't much point checking
+                    # for deprecated usage as there is no guarantee as to the
+                    # state of the policy.
+                    if options.check_deprecations != "skip":
+                        report = deprecated.report(policy)
+                        if report:
+                            found_deprecations = True
+                            # TODO: consider different formats for output.
```

Comment:
consideration of different formats would require aggregation of the reports across policies.

Reply:
Perhaps worth removing the TODO. The purpose of the footnotes is to aggregate information across policies, so we only get the extra information around deprecation deadline and documentation once for the validation run.